packages/block-library/src/template-part/edit/index.js

```diff
@@ -78,6 +82,17 @@ export default function TemplatePartEdit( {
 		);
 	}
 
+	if ( hasAlreadyRendered ) {
```

Comment:
What happens when `! theme || ! slug`?

Reply:
I haven't tested this, but my concern is that `templatePartId` depends on those two variables, and passing a null value to `useNoRecursiveRenders` is something to approach with care, because `hasAlreadyRendered` might come out true even though that template part is not recursive.

Reply:
Yes, it's on my mind. See https://github.com/WordPress/gutenberg/pull/28456#discussion_r563731762 that is collapsed now. It's not clear to me why one of them would be `null` which makes it difficult to process in the first place.

Reply:
I opted for:
```
if ( isEntityAvailable && hasAlreadyRendered ) {
```

`isEntityAvailable` is true only when inner blocks are ready to render. It covers also the case when `templatePartId` is `null`  which means there is no way to find an entity.